### PR TITLE
chore(components): Fix typos in Great Expectations component

### PR DESCRIPTION
**Pull request by strawberrypie**
**Description of your changes:**

This PR fixes a couple of typos introduced in #5235.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: components/great-expectations/validate/CSV/component.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 def validate_csv_using_greatexpectations(
     csv_path: InputPath(),
-    expectation_suite_path: InputPath(),
+    expectation_suite_path: str,
```
Comment (NikeNano) [inline components/great-expectations/validate/CSV/component.py:6]: why is this change needed?

Comment (strawberrypie) [reply components/great-expectations/validate/CSV/component.py:6]: It is needed to correctly pass pipeline arguments to this component. They are passed as file in case of `InputPath`.

Comment (Ark-kun) [reply components/great-expectations/validate/CSV/component.py:6]: What kind of "path" is this? Is it a URI? Let's be clear.
`expectation_suite_uri`

File: components/great-expectations/validate/CSV/component.py [hunk 2]
```diff
@@ -42,9 +44,10 @@ def validate_csv_using_greatexpectations(
 
 
 if __name__ == '__main__':
-    calculate_regression_metrics_from_csv_op = create_component_from_func(
+    validate_csv_using_greatexpectations = create_component_from_func(
```
Comment (Ark-kun) [inline components/great-expectations/validate/CSV/component.py:47]: Are you sure? The function returned from `create_component_from_func` is slightly different from the one passed in. It might be better to give them different names.

File: components/great-expectations/validate/CSV/component.py (hunk 1 above)
Comment (Ark-kun) [reply components/great-expectations/validate/CSV/component.py:6]: Your original version has some advantage over this new version.
With the old version I can process/transform a dataset inside the pipeline and then run the validation. With the proposed change, I can no longer do that - I need to first upload the dataset to some cloud storage (Which one? How do I authenticate?)

**Comment by Ark-kun** [comment]
Your original version has some advantage over this proposed new version.

With the old version I can process/transform a dataset inside the pipeline and then run the validation. With the proposed change, I can no longer do that - I need to first upload the dataset to some cloud storage. (Which one? How do I authenticate?)

This proposal combines downloading and validation into a single component and it seems to reduce the flexibility.
Can you explain the scenario (a pipeline) where you want a step which validates a CSV dataset at some explicit URI? What would be an upstream of such component? What would be a downstream?

If you want a URI-based version, perhaps you can add that as a separate component in a `from_URI` subfolder. What do you think?

File: components/great-expectations/validate/CSV/component.py (hunk 1 above)
Comment (NikeNano) [reply components/great-expectations/validate/CSV/component.py:6]: Maybe it is better to download the Great Exception `suite` in a separate step in which `open` or any other package to fetch it could be use, similar to how it is set up today? That would result in two steps in a pipeline but give the user a lot more freedom.

Comment (strawberrypie) [reply components/great-expectations/validate/CSV/component.py:6]: @NikeNano @Ark-kun sure, you're right, thanks for the review. I'l rollback this change and this PR will fix only a couple of typos that we're left from the previous PR.

File: components/great-expectations/validate/CSV/component.py (hunk 2 above)
Comment (strawberrypie) [reply components/great-expectations/validate/CSV/component.py]: Sure. The variable name isn't important here because it's not used, but we need to use different names for pipeline steps and functions for consistency. I've change the variable name to a more appropriate one.

**Comment by strawberrypie** [comment]
@NikeNano @Ark-kun thanks for the review. The idea is that Expectation Suites aren't changed that much and it makes sense to store them in a place like S3. However, I agree that there are many possible scenarios and the one implemented here (ExpectationSuite is passed as an input rather than a path) is the most universal. I rolled back the suggested change with smart_open.

**Review by NikeNano — APPROVED** [review]
/LGTM

**Comment by Ark-kun** [comment]
>The idea is that Expectation Suites aren't changed that much and it makes sense to store them in a place like S3.
>However, I agree that there are many possible scenarios and the one implemented here (ExpectationSuite is passed as an input rather than a path) is the most universal. I rolled back the suggested change with smart_open.

I apologize if my tone sounded too strict.
As the author of the component you're free to design your components however you please.
What I said was meant more as an advice than a requirement.
As the author, you understand the use cases of the component much better than me.

Using the built-in file-based data-passing is best practice, which makes components simpler and ensures portability.
However I also understand that in some cases it becomes harder to use (e.g. when accessing 100GB+ datasets), in which case it makes sense to create a URI-based version of a component to handle that special use case (although the file-based version should probably remain for more normal use cases).
